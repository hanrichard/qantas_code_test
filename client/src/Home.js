import React from 'react';
import Container from '@material-ui/core/Container';
import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';

export const GET_AIRPORTS_QUERY = gql`
    {
        airports {
            airportCode
            airportName
            location {
                latitude
                longitude
            }
            city {
                timeZoneName
            }
            country {
                countryName
            }
        }
    }
`;

const Home = () => {
    const { loading, error, data } = useQuery(GET_AIRPORTS_QUERY);

    if (loading) return <div>loading...</div>;
    if (error) return <div>{error.message}</div>;

    console.log(data.airports);

    return (
        <div>
            <div>
                <Container maxWidth="sm">
                    <div className="AirportList__list">list</div>
                </Container>
            </div>
        </div>
    );
};

export default Home;
