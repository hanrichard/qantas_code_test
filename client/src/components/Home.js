import React, { useState } from 'react';
import AirportItem from './AirportItem';
import styled from 'styled-components';
import componentStyle from './Home.style';
import Pagination from 'react-js-pagination';
import Container from '@material-ui/core/Container';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { pageShownNum, pageRangeDisplayed, currentPages, nextPages, totalNumber } from '../utility/utility';

const Wrapper = styled.div`
    ${componentStyle}
`;
const Home = ({ airports }) => {
    const [activePage, setActivePage] = useState(1);

    const airportsList =
        airports &&
        airports.slice(currentPages(activePage), nextPages(activePage)).map(airport => {
            return <AirportItem key={airport.airportCode} airport={airport} />;
        });

    return (
        <Wrapper>
            {airports && airports.length > 0 ? (
                <div>
                    <div className="AirportList__list">{airportsList}</div>

                    <div className="AirportList__pagination">
                        <Container maxWidth="sm">
                            <Pagination
                                activePage={activePage}
                                itemsCountPerPage={pageShownNum}
                                totalItemsCount={totalNumber(airports) - pageShownNum}
                                pageRangeDisplayed={pageRangeDisplayed}
                                onChange={activePage => setActivePage(activePage)}
                            />
                        </Container>
                    </div>
                </div>
            ) : (
                <div>
                    <Typography variant="h4" component="h1" gutterBottom>
                        No result
                    </Typography>
                </div>
            )}
        </Wrapper>
    );
};

Home.propTypes = {
    airport: PropTypes.shape({
        airportName: PropTypes.string,
        airportCode: PropTypes.string,
        country: PropTypes.shape({
            countryName: PropTypes.string,
        }),
    }),
};

export default Home;
