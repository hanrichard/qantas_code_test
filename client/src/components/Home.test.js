import React from 'react';
import { shallow, configure } from 'enzyme';
import Home from './Home';
import Adapter from 'enzyme-adapter-react-16';
import Renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

function setup() {
    const props = {
        airports: [
            {
                airportCode: 'AAA',
                airportName: 'Anaa',
                location: {
                    latitude: '17.25',
                    longitude: '145.3',
                },
                city: {
                    timeZoneName: 'Pacific/Tahiti',
                },
                country: {
                    countryName: 'French Polynesia',
                },
            },
        ],
    };

    const shallowWrapper = shallow(<Home {...props} data-test="test" />);

    return {
        props,
        shallowWrapper,
    };
}

describe('Airport Home', () => {
    it('should match snapshot', () => {
        const airportHome = Renderer.create(`<BrowserRouter>shallowWrapper</BrowserRouter>`).toJSON();
        expect(airportHome).toMatchSnapshot();
    });
});

describe('Home component', () => {
    const { shallowWrapper } = setup();

    it('should render', () => {
        expect(shallowWrapper.find('.AirportList__list')).toHaveLength(1);
        expect(shallowWrapper.find('.AirportList__pagination')).toHaveLength(1);
    });

    it('should render', () => {
        shallowWrapper.setProps({ airports: [] });
        expect(shallowWrapper.find('.AirportList__list')).toHaveLength(0);
        expect(shallowWrapper.find('.AirportList__pagination')).toHaveLength(0);
        expect(shallowWrapper.text().includes('No result')).toBe(true);
    });
});
