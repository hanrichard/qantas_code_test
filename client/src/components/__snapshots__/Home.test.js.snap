// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airport Home should match snapshot 1`] = `"<BrowserRouter>shallowWrapper</BrowserRouter>"`;
